parse_cmd_test.go

```diff
@@ -1,151 +1,134 @@
 package main
 
 import (
+	"errors"
+	"reflect"
 	"testing"
 )
 
-var daysList = map[string]struct{}{
-	"monday":    {},
-	"tuesday":   {},
-	"wednesday": {},
-	"thursday":  {},
-	"friday":    {},
-	"saturday":  {},
-	"sunday":    {},
+type parseResult struct {
+	Cmd  string
+	Args []string
 }
 
-// due to the difficulties around comparing error return values (and because we don't want to compare error messages),
-// the struct contains expectErr to indicate whether an error is expected, instead of an actual error value
-var tableNoArgs = []struct {
-	testName   string
-	inputStr   string
-	wantedCmd  string
-	wantedArgs []string
-	expectErr  bool
-}{
-	{"subscribe_correct_usage", "subscribe", "subscribe", nil, false},
-	{"subscribe_wrong_usage", "subscribe mon", "help", nil, true},
-	{"unsubscribe_correct_usage", "unsubscribe", "unsubscribe", nil, false},
-	{"unsubscribe_wrong_usage", "unsubscribe tuesday", "help", nil, true},
-	{"help_correct_usage", "help", "help", nil, false},
-	{"help_wrong_usage", "help me", "help", nil, true},
-	{"status_correct_usage", "status", "status", nil, false},
-	{"status_wrong_usage", "status me", "help", nil, true},
-	{"version", "version", "version", nil, false},
-}
+var acceptedCommands = map[string]parseResult{
+	"subscribe":   {"subscribe", nil},
+	"unsubscribe": {"unsubscribe", nil},
+	"help":        {"help", nil},
+	"status":      {"status", nil},
+	"get-reviews": {"get-reviews", nil},
+	"cookie":      {"cookie", nil},
+	"version":     {"version", nil},
 
-func TestParseCmdNoArgs(t *testing.T) {
-	for _, tt := range tableNoArgs {
-		t.Run(tt.testName, func(t *testing.T) {
-			gotCmd, gotArgs, gotErr := parseCmd(tt.inputStr)
-			if gotCmd != tt.wantedCmd {
-				t.Errorf("got %v, %v\n", gotCmd, gotArgs)
-			}
+	// This command ignores its arguments.
+	"version info": {"version", nil},
 
-			_, ok := gotErr.(*parsingErr)
+	// These commands require exact literal arguments.
+	"skip tomorrow":   {"skip", []string{"tomorrow"}},
+	"unskip tomorrow": {"unskip", []string{"tomorrow"}},
 
-			if tt.expectErr && !ok {
-				t.Errorf("Expected parsingErr but didn't get one\n")
-			} else if !tt.expectErr && ok {
-				t.Errorf("Got unexpected parsingError\n")
-			}
-		})
-	}
+	// Schedules!
+	"schedule monday":         {"schedule", []string{"monday"}},
+	"schedule sunday":         {"schedule", []string{"sunday"}},
+	"schedule friday tuesday": {"schedule", []string{"friday", "tuesday"}},
+	"schedule mon tue wed thu fri sat sun": {
+		"schedule",
+		[]string{"monday", "tuesday", "wednesday", "thursday", "friday", "saturday", "sunday"},
+	},
+
+	// BUG(@jdkaplan): Don't squash spaces *inside* the review.
+	"add-review  :pear: ing    :robot:": {"add-review", []string{":pear: ing :robot:"}},
+
+	"get-reviews 0":  {"get-reviews", []string{"0"}},
+	"get-reviews 1":  {"get-reviews", []string{"1"}},
+	"get-reviews 5":  {"get-reviews", []string{"5"}},
+	"get-reviews 10": {"get-reviews", []string{"10"}},
```

Comment:
I want to argue that `get-reviews 0` is a bug, but it has funny output, so I'm keeping it :smile: